backend/src/main/java/pl/edu/pw/ee/backend/api/dish/BazaPosilkow.java

```diff
@@ -34,17 +31,34 @@ public class BazaPosilkow implements IBazaPosilkow {
     private final ImageServiceImpl imageService;
 
     @Override
-    public List<FindDishDTO> getByFiltr(int clientId, FiltrDTO filtrObject) {
-        log.debug("Getting dishes for client {} with filters: {}", clientId, filtrObject);
+    public List<FindDishDTO> getByFiltr(int clientId, FilterDTO filterObject) {
+        log.debug("Getting dishes for client {} with filters: {}", clientId, filterObject);
 
         if (!clientRepository.existsById(clientId)) {
             throw new ClientNotFoundException(clientId);
         }
 
+        List<MealType> mealTypes = filterObject.mealFilter() != null ?
+                filterObject.mealFilter().values() : null;
+        List<String> kitchenTypes = filterObject.kitchenFilter() != null ?
+                filterObject.kitchenFilter().values() : null;
+        List<String> companyNames = filterObject.companyFilter() != null ?
+                filterObject.companyFilter().values() : null;
+
+        String mealSorting = filterObject.mealFilter() != null ?
+                filterObject.mealFilter().sorting().name() : null;
+        String kitchenSorting = filterObject.kitchenFilter() != null ?
+                filterObject.kitchenFilter().sorting().name() : null;
+        String companySorting = filterObject.companyFilter() != null ?
+                filterObject.companyFilter().sorting().name() : null;
+
         List<Dish> dishes = dishRepository.findDishesWithFilters(
```

Comment:
Brakuje tutaj Klasy separującej repozytoria od IBazaPosilkow. Tutaj powinieneś dać jaki IDishRepository. Sprawdź jak to jest robione w innych uckach na branchu devel

Reply:
@czaacza taka klasa już jest dodaj do niej tę metodę de facto będziesz w niej callował metodę z repozytorium